### PR TITLE
Support login via username or email address for OpenID

### LINKED ISSUE #10190: Can't get access token via email address as user name
### To Reproduce
Get access token via POST https://localhost:44391/connect/token
- grant_type
- scope
- username
- password
- client_id

If I use the user name, It works. 
However, if I use user email. It doesn't work. 

### Expected behavior
-  According to this discussion thread: https://github.com/OrchardCMS/OrchardCore/pull/7645. There is a change to not allow the username = email. However, there should be a fallback to authenticate by username or email. 
-  I expect I can use username or email as to login. 

### Screenshots
image

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #10190

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs [hunk 1]
```diff
@@ -97,7 +108,7 @@ public async Task<IActionResult> Authorize()
                     {
                         [OpenIddictServerAspNetCoreConstants.Properties.Error] = Errors.ConsentRequired,
                         [OpenIddictServerAspNetCoreConstants.Properties.ErrorDescription] =
-                            "The logged in user is not allowed to access this client application."
+                            S["The logged in user is not allowed to access this client application."]
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs:111]: ⚠️ 

https://github.com/OrchardCMS/OrchardCore/blob/19f34caa1bf9b6c633146da80e340ff119424e15/src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs#L30-L31

File: src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs [hunk 2]
```diff
@@ -490,7 +501,17 @@ private async Task<IActionResult> ExchangePasswordGrantType(OpenIddictRequest re
                 {
                     [OpenIddictServerAspNetCoreConstants.Properties.Error] = Errors.UnsupportedGrantType,
                     [OpenIddictServerAspNetCoreConstants.Properties.ErrorDescription] =
-                        "The resource owner password credentials grant is not supported."
+                        S["The resource owner password credentials grant is not supported."]
+                }), OpenIddictServerAspNetCoreDefaults.AuthenticationScheme);
+            }
+
+            var disableLocalLogin = (await _siteService.GetSiteSettingsAsync()).As<LoginSettings>().DisableLocalLogin;
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs:508]: We should probably have that logic directly in `UserService.AuthenticateAsync()`.

File: src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj [hunk 3]
```diff
@@ -26,6 +26,7 @@
     <ProjectReference Include="..\..\OrchardCore\OrchardCore.Recipes.Abstractions\OrchardCore.Recipes.Abstractions.csproj" />
     <ProjectReference Include="..\..\OrchardCore\OrchardCore.ResourceManagement\OrchardCore.ResourceManagement.csproj" />
     <ProjectReference Include="..\..\OrchardCore\OrchardCore.Users.Abstractions\OrchardCore.Users.Abstractions.csproj" />
+    <ProjectReference Include="..\..\OrchardCore\OrchardCore.Users.Core\OrchardCore.Users.Core.csproj" />
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj:29]: Why?

**Review by kevinchalet — CHANGES_REQUESTED** [review]
Thanks for your PR.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs (hunk 2 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs:508]: Should we? What if we want to authenticate using other services while the local login in disabled?

File: src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj (hunk 3 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj:29]: To be able to read the `LoginSettings` settings which is where it is located.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs (hunk 1 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs]: Thanks. I undid this.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs (hunk 2 above)
Comment (kevinchalet) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs:503]: This method is only used for local logins (it's backed by Identity's `CheckPasswordSignInAsync()`, which is a password-only thing anyway).

File: src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj (hunk 3 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj]: @kevinchalet done!

**Comment by kevinchalet** [comment]
Merged, thanks! 👏🏻

**Pull request merged**